apps/prairielearn/src/schemas/schemas/infoAssessment.ts

```diff
@@ -0,0 +1,371 @@
+  .number()
+  .gte(0)
+  .lte(100)
+  .describe('Minimum score percentage to unlock access to subsequent questions');
+
+const QuestionPointsSchema = z.object({
+  points: PointsSchema.optional(),
+  autoPoints: PointsSchema.optional(),
+  maxPoints: PointsSingleSchema.optional(),
+  maxAutoPoints: PointsSingleSchema.optional(),
+  manualPoints: PointsSingleSchema.optional(),
+});
+
+export type QuestionPoints = z.infer<typeof QuestionPointsSchema>;
+
+const QuestionAlternativeSchema = QuestionPointsSchema.extend({
+  comment: CommentSchema.optional(),
+  id: QuestionIdSchema,
+  forceMaxPoints: ForceMaxPointsSchema.optional(),
+  triesPerVariant: z
+    .number()
+    .describe('The maximum number of graded submissions allowed for each question instance.')
+    .optional(),
+  advanceScorePerc: AdvanceScorePercSchema.optional(),
+  gradeRateMinutes: z
+    .number()
+    .gte(0)
+    .describe(
+      'Minimum amount of time (in minutes) between graded submissions to the same question.',
+    )
+    .optional(),
+  canView: uniqueArray(z.string())
+    .describe('The names of roles that can view this question.')
+    .optional(),
+  canSubmit: uniqueArray(z.string())
+    .describe('The names of roles that can submit this question.')
+    .optional(),
+});
+
+export const ZoneQuestionSchema = QuestionPointsSchema.extend({
+  comment: CommentSchema.optional(),
+  points: PointsSchema.optional(),
+  autoPoints: PointsSchema.optional(),
+  maxPoints: PointsSingleSchema.optional(),
+  maxAutoPoints: PointsSingleSchema.optional(),
+  manualPoints: PointsSingleSchema.optional(),
+  id: QuestionIdSchema.optional(),
+  forceMaxPoints: ForceMaxPointsSchema.optional(),
+  alternatives: z
+    .array(QuestionAlternativeSchema)
+    .min(1)
+    .describe('Array of question alternatives to choose from.')
+    .optional(),
+  numberChoose: z
+    .number()
+    .int()
+    .gte(0)
+    .describe('Number of questions to choose from this group.')
+    .optional(),
+  triesPerVariant: z
+    .number()
+    .describe('The maximum number of graded submissions allowed for each question instance.')
+    .optional(),
+  advanceScorePerc: AdvanceScorePercSchema.optional(),
+  singleVariant: z
+    .boolean()
+    .describe(
+      'Whether the question is not randomized and only generates a single variant (defaults to "false").',
+    )
+    .optional(),
+  gradeRateMinutes: z
+    .number()
+    .gte(0)
+    .describe(
+      'Minimum amount of time (in minutes) between graded submissions to the same question.',
+    )
+    .optional(),
+  canSubmit: uniqueArray(z.string())
+    .describe(
+      'A list of group role names matching those in groupRoles that can submit the question. Only applicable for group assessments.',
+    )
+    .optional(),
+  canView: uniqueArray(z.string()).describe(
+    'A list of group role names matching those in groupRoles that can view the question. Only applicable for group assessments.',
+  ),
```

Comment:
This needs to be `.optional()`. Otherwise basically every assessment fails to sync.

Have you updated our syncing code to use these new schemas yet? I'm kind of shocked that our tests didn't catch this.

Reply:
> This needs to be .optional(). Ditto in the other schemas in this file. Otherwise basically every assessment fails to sync.

Made optional in [6421c73](https://github.com/PrairieLearn/PrairieLearn/pull/11024/commits/6421c738649ad34889d1adc1dcde6ca993d376dc).

Reply:
> Have you updated our syncing code to use these new schemas yet? I'm kind of shocked that our tests didn't catch this.

I have not. A lot of the syncing code still uses types from `course-db`.

Reply:
I'm talking about schemas, not types. Yes, we should use them wherever the old schemas were used: that's both a valuable test that they're working correctly, and also these schemas really only exist for syncing, so they're not doing us any good if they're not used there.

Reply:
Oh I misunderstood -- yes the new schemas are used for syncing. Maybe I am still misunderstanding, how would it be possible to still use the old schemas if I have ripped them out?

Reply:
TBH, all I was looking at were the unexpected passing CI checks, but when I looked at them more closely (e.g. https://github.com/PrairieLearn/PrairieLearn/actions/runs/12663921484) it looks like the actual tests weren't running. Now they are, and there are correctly a bunch of failing tests.

Reply:
This should be fixed now, and running through our actual test suite.